smart-contracts/assembly/contracts/NFT/NFTEnumerable-example.ts

```diff
@@ -0,0 +1,253 @@
+  const args = new Args(binaryArgs);
+  const owner = args
+    .nextString()
+    .expect('owner argument is missing or invalid');
+  const operator = args
+    .nextString()
+    .expect('operator argument is missing or invalid');
+  return boolToByte(_isApprovedForAll(owner, operator));
+}
+
+/**
+ *
+ * @param binaryArgs - serialized strings representing the address of the recipient and the tokenId to approve
+ * @remarks This function is only callable by the owner of the tokenId or an approved operator.
+ * Indeed, this will be checked by the _approve function of the NFT-internals.
+ *
+ */
+export function approve(binaryArgs: StaticArray<u8>): void {
+  const args = new Args(binaryArgs);
+  const to = args.nextString().expect('to argument is missing or invalid');
+  const tokenId = args
+    .nextU256()
+    .expect('tokenId argument is missing or invalid');
+  _approve(to, tokenId);
+}
+
+/**
+ *
+ * @param binaryArgs - serialized arguments representing the address of the operator and a boolean value indicating
+ * if the operator should be approved for all the caller's tokens
+ *
+ */
+export function setApprovalForAll(binaryArgs: StaticArray<u8>): void {
+  const args = new Args(binaryArgs);
+  const to = args.nextString().expect('to argument is missing or invalid');
+  const approved = args
+    .nextBool()
+    .expect('approved argument is missing or invalid');
+  _setApprovalForAll(to, approved);
+}
+
+/**
+ *
+ * @param binaryArgs - serialized arguments representing the address of the sender,
+ * the address of the recipient, and the tokenId to transfer.
+ *
+ * @remarks This function is only callable by the owner of the tokenId or an approved operator.
+ */
+export function transferFrom(binaryArgs: StaticArray<u8>): void {
+  const args = new Args(binaryArgs);
+  const from = args.nextString().expect('from argument is missing or invalid');
+  const to = args.nextString().expect('to argument is missing or invalid');
+  const tokenId = args
+    .nextU256()
+    .expect('tokenId argument is missing or invalid');
+  _transferFrom(from, to, tokenId);
+}
+
+/**
+ *
+ * @param binaryArgs - serialized arguments representing the address of the recipient and the tokenId to mint
+ *
+ * @remarks This function is only callable by the owner of the contract.
+ */
+export function mint(binaryArgs: StaticArray<u8>): void {
+  onlyOwner();
+  const args = new Args(binaryArgs);
+  const to = args.nextString().expect('to argument is missing or invalid');
+  const tokenId = args
+    .nextU256()
+    .expect('tokenId argument is missing or invalid');
+  _update(to, tokenId, '');
+}
+
+/**
+ *
+ * @param binaryArgs - serialized u256 representing the tokenId to burn
+ *
+ * @remarks This function is not part of the ERC721 standard.
+ * It serves as an example of how to use the NFT-enumerable-internals functions to implement custom features.
+ */
+export function burn(binaryArgs: StaticArray<u8>): void {
```

Comment:
anyone can burn any token ?

Reply:
The check is done in updateBase (internals)